src/main/java/com/artipie/settings/YamlSettings.java

```diff
@@ -214,9 +168,10 @@ private static CachedUsers initAuth(final YamlMapping settings) {
             );
             res = new AuthFromEnv();
         } else {
+            final AuthLoader loader = new AuthLoader();
             final List<Authentication> auth = creds.values().stream().map(
-                node -> CredentialsType.valueOf(node.asMapping())
-            ).map(type -> type.auth(settings)).toList();
+                node -> node.asMapping().string(YamlSettings.NODE_TYPE)
+            ).map(type -> loader.newObject(type, settings)).toList();
             res = auth.get(0);
```

Comment:
Can we add check of size of 'auth' list? Because if auth is empty then IndexOutOfBoundsException will be thrown.

Reply:
hm, it's not possible as we checked credentials sequence and it's not empty. Then, we perform mapping.